### PR TITLE
[autoware state monitor] support autoware.auto msg

### PR DESCRIPTION
**Pull request by tkimura4**
## Description

support .auto msg
autoware_planning_msgs/msg/route -> autoware_auto_planning_msgs/msg/route

autoware_planning_msgs/msg/trajectory -> autoware_auto_planning_msgs/msg/trajectory

autoware_system_msgs/msg/autoware_state -> autoware_auto_system_msgs/msg/autoware_state

autoware_system_msgs/msg/engage -> autoware_auto_system_msgs/msg/engage

autoware_system_msgs/msg/hazard_status_stamped -> autoware_auto_system_msgs/msg/hazard_status_stamped

autoware_vehicle_msgs/msg/control_mode -> autoware_auto_vehicle_msgs/msg/vehicle_state_report

geometry_msgs/msg/twist_stamped -> nav_msgs/msg/odomtery

- add readme (from scratch)

### Remaining Issues
~- Handling of string message in original AutowareState.msg~
~- Handling of Emergency state in original AutowareState.msg~

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp [hunk 1]
```diff
@@ -110,35 +112,37 @@ std::string getStateMessage(const AutowareState & state)
 
   throw std::runtime_error("invalid state");
 }
+*/
```
Comment (kenji-miyake) [inline system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:115]: Remove?

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp [hunk 2]
```diff
@@ -269,19 +273,8 @@ void AutowareStateMonitorNode::onTimer()
...
-    // Add messages line by line
-    std::ostringstream oss;
-
-    oss << getStateMessage(autoware_state) << std::endl;
-
-    for (const auto & msg : state_machine_->getMessages()) {
-      oss << msg << std::endl;
-    }
-
-    autoware_state_msg.msg = oss.str();
```
Comment (kenji-miyake) [inline system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:284]: Could you remove `state.msg` from autoware_web_controller?

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp (hunk 1 above)
Comment (tkimura4) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:115]: Is it okay to completely delete the code related to the string-type message in original AutowareState.msg?

Comment (kenji-miyake) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:115]: Do you mean this? Then, it's okay!
https://github.com/tier4/autoware_iv_msgs/blob/0477e88d1da64df4497b002c7793f8c619ffba52/autoware_system_msgs/msg/AutowareState.msg#L11

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp (hunk 2 above)
Comment (tkimura4) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:284]: in this PR?

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp (hunk 1 above)
Comment (tkimura4) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:115]: yes, Thanks.

Comment (tkimura4) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp]: Fixed at 9905e73

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp (hunk 2 above)
Comment (kenji-miyake) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:284]: Or another PR is okay. I'll leave it to you!

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp (hunk 1 above)
Comment (tkimura4) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp]: Do I need to remove the code related to EmergecyState as well?

Comment (kenji-miyake) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp]: Thank you :heart:

File: system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp (hunk 2 above)
Comment (tkimura4) [reply system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp:284]: autoware_web_controller seems to have many other parts that need to be modified, so I will divide the PR

**Comment by tkimura4** [comment]
@kenji-miyake 
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/521#discussion_r743413365
Do I need to remove the code related to EmergecyState as well?

(In autoware.auto, AutowareState has not a emergency state. )

**Comment by kenji-miyake** [comment]
I think you should remove Emergency-related code.

**Comment by tkimura4** [comment]
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/521#issuecomment-961656691
Done.
56bdc64
4f4150d

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**